tests/jobs/scala/src/main/scala/KafkaCassandraStream.scala

```diff
@@ -0,0 +1,385 @@
+            email_address text,
+            address text,
+            city text
+          );"""
+        )
+      }
+
+      val recordsRdd = normalRDD(spark.sparkContext, numberOfRecords, numberOfPartitions).map(_ => randomEmployee(minColumnsLength, maxColumnsLength))
+      // TODO: input stream source should be Kafka.
+      val recordsStream = new ConstantInputDStream(streamingContext, recordsRdd)
+      val employeeColumns = SomeColumns("id", "first_name", "last_name", "email_address", "address", "city")
+
+      recordsStream.foreachRDD { rdd =>
+        println(s"Writing batch of size ${rdd.count} to Cassandra...")
+        val t0 = System.nanoTime()
+        rdd.saveToCassandra(keyspaceName, tableName, employeeColumns)
+        val t1 = System.nanoTime()
+        println(s"Wrote batch in ${(t1 - t0) / 1000 / 1000} ms")
+
+        // TODO: remove this when not using ConstantInputDStream anymore.
+        streamingContext.stop() // we only need to go through the ConstantInputDStream once.
+      }
+
+      streamingContext.start()
+      streamingContext.awaitTermination()
+    }
+  }
+}
+
+
+object KafkaRandomFeeder {
+  def main(args: Array[String]): Unit = {
+
+    val config : Config = Config()
+
+    if (!getParser("KafkaRandomFeeder").parse(args, config).isDefined) {
+      println("Error: Bad arguments")
+      System.exit(1)
+    }
+
+    val sparkBuilder = SparkSession.builder().appName(config.appName)
+
+    val spark = sparkBuilder
+      .master("local[2]")
+      .getOrCreate()
+
+    val streamingContext = new StreamingContext(spark.sparkContext, Seconds(config.batchSizeSeconds))
+
+    val wordsPerSecond = config.wordsPerSecond
+    val stream = streamingContext.receiverStream(new RandomWordReceiver(wordsPerSecond))
+
+
+    val kafkaProperties = getKafkaProperties(config.brokers, config.kerberized)
+    val topic : String = config.topics
+
+    stream.foreachRDD { rdd =>
+      rdd.foreachPartition { p =>
+
+        val producer = new KafkaProducer[String, String](kafkaProperties)
+        p.foreach { r =>
+          println(s"Writing ${r} to Kafka")
+          val msg = new ProducerRecord[String, String](topic, null, r)
+          producer.send(msg)
+        }
+        producer.close()
+
+      }
+    }
+
+    streamingContext.start()
+    streamingContext.awaitTermination()
+  }
+
+
+  def getKafkaProperties(brokers : String, kerberized: Boolean) : Properties = {
+
+    val props = new Properties()
+    props.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, brokers)
+    props.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, "org.apache.kafka.common.serialization.StringSerializer")
+    props.put(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, "org.apache.kafka.common.serialization.StringSerializer")
+    if (kerberized) {
+      props.put("sasl.kerberos.service.name", "kafka")
+      props.put("security.protocol", "SASL_PLAINTEXT")
+      props.put("sasl.mechanism", "GSSAPI")
+    }
+
+    return props
```

Comment:
`return` keyword is frowned upon in Scala: https://tpolecat.github.io/2014/05/09/return.html